core/app/models/spree/tax/tax_location.rb

```diff
@@ -0,0 +1,35 @@
+module Spree
+  module Tax
+    # A class exclusively used as a drop-in replacement for a default tax address.
+    # It responds to `:country_id` and `:state_id`.
+    #
+    # @attr_reader [Integer] country_id the ID of a Spree::Country object
+    # @attr_reader [Integer] state_id the ID of a Spree::State object
+    class TaxLocation
+      attr_reader :country_id, :state_id
+
+      # Create a new TaxLocation object
+      #
+      # @see Spree::Zone.for_address
+      #
+      # @param [Integer] country_id the ID of a Spree::Country object, default: nil
+      # @param [Integer] state_id the ID of a Spree::State object, default: nil
+      # @param [Spree::Country] country a Spree::Country object, default: nil
+      # @param [Spree::State] state a Spree::State object, default: nil
+      #
+      # @return [Spree::Tax::TaxLocation] a Spree::Tax::TaxLocation object
+      def initialize(country: nil, country_id: nil, state: nil, state_id: nil)
```

Comment:
I would prefer that we deal strictly with objects rather than objects or id's. It makes things a bit simpler. It can likely be more efficient by just using the ID's, but I'd prefer the simpler code for now.